app/db/interface.py

```diff
@@ -66,5 +67,27 @@ def get_payments(self, payment_ids=None, payment_attributes=None):
         :param payment_attributes: <dict> a collection of payment attributes to be used in
                                    filtering for specific payments
         """
+        if payment_ids != None:
+            payments = self.db.session.query(Payment).filter(Payment.id.in_(payment_ids)).all()
```

Comment:
I think you should be able to use the internal _query_payments method.  It's a little awkward but this would work:

```
id_attribute_dicts = [{'payment_id': id} for id in payment_ids]
payments = [self._query_payments(payment_attributes=attributes)[0] for key, attributes in id_attribute_dicts.iteritems()]
```

We need to index by zero since each call *should* return a list with a single payment element.
Also, it'd be best to wrap this guy in a try/except block.